lib/Sema/CSSimplify.cpp

```diff
@@ -3286,6 +3310,40 @@ performMemberLookup(ConstraintKind constraintKind, DeclName memberName,
       }
     }
   }
+  
+  // If we're about to fail lookup, but we are looking for members in a type
+  // that conforms to DynamicMemberLookupProtocol, then we resolve the reference
+  // to the subscript(dynamicMember:) member, and pass the member name as a
+  // string.
+  if (constraintKind == ConstraintKind::ValueMember && hasInstanceMembers &&
+      memberName.isSimpleName() && !memberName.isSpecial()) {
+    auto name = memberName.getBaseIdentifier();
+    if (containsOrConformsToDLMP(instanceTy, DC, name, TC)) {
+      // Recursively look up the subscript(dynamicMember:)'s in this type.
+      auto subscriptName =
+        DeclName(getASTContext(), DeclBaseName::createSubscript(),
```

Comment:
A few places in this method use TC.Context and here you're using getASTContext(), might want to just extract a ctx local variable